### PR TITLE
[Entry Term] Neural Networks ai adam optimization

### PR DESCRIPTION
**Pull request by TimLyons87**
### Description

- Added Adam Optimization term for the AI Neural Networks
- Closes Issue #4528 
### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by mamtawardhani** [comment]
Hey @TimLyons87! The submitted entry is empty, could you please check?

**Comment by TimLyons87** [comment]
@mamtawardhani adam-optimization.md is in my repositorties' main branch. I have attempted to change it to the correct repository and it should be ready for review now. Thank you very much for your continued patience.

**Comment by mamtawardhani** [comment]
@TimLyons87 I can still see an empty entry, you need to create a feature branch for this, it should not be on the main branch. I think you can raise another Pull request by adding the content on the featured branch.

**Comment by TimLyons87** [comment]
@mamtawardhani thank you again for your patience. I have created a new pull request as instructed. The files should be visible now and ready for review. Thanks, Tim.

File: content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+## Challenges
+[While very robust Adam does have a few challenges.
+1. Adam does not always converge to an optimal solution
+2. Adam does suffer from some weight decay
+3. Newer optimization algorithms are faster and provide improved calculations]
+
+## Example
+
+[Below is an example of the pytorch Adam optimizer syntax that is used to reduce the rate of error during the training of the target neural network.
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md:65]: These lines can be removed from here

File: content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ '''python
+ optimizer = torch.optim.Adam(params,lr=0.005,betas=0.9,0.999,eps=1e-08,weight_decay=0)'''
+
+ The class statement and the required arguments used in the syntax are outlined below:
+ - class torch.optim.Adam identifies Adam as the specific optimizer for gradient descent.
+ - params: used as a parameter that helps in the optimization.
+ - lr: learning rate helping the optimizer.
+ - betas: decay parameter that calculates the running average of the gradient and its square.
+ - eps: epsilon term added to denominator used to improve numerical stability.
+ - weight_delay: adds an L2 penalty to the loss function with a default value of 0.
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md:77]: Use the below format to describe parameters:
- `params`: description
- `lr`: description
and other parameters...

File: content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+Adam optimization is used for sparse and noisy datasets like those found in large language models or image processing models.]
+
+## Codebyte Example (if applicable)
+[Codebyte example is provided in numpy to better show steps in optimization. The source code is from deeplearning.ai]
+
+```codebyte/python
+# Example runnable code block.
+ [# import libraries](https://www.codecademy.com/workspaces/664e05e18948c76448d270d9);
+```
```
Comment (mamtawardhani) [inline content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md:87]: Use and ## Example section to show an example of the `torch.optim.Adam` function you have described earlier, and also have a ## Codebyte Example if you can include it.

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @TimLyons87! Thank you for Contributing to Codecademy Docs! The entry was excellent!

I've suggested a few changes for the entry, could you please review and modify them at your earliest convenience? Feel free to discuss anything 😄 
Thank you!

File: content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md (hunk 1 above)
Comment (TimLyons87) [reply content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md:66]: Lines deleted

File: content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md (hunk 2 above)
Comment (TimLyons87) [reply content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md:72]: Updated syntax section to reflect requested format changes.

File: content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md (hunk 3 above)
Comment (TimLyons87) [reply content/ai/concepts/neural-networks/terms/Adam-Optimization/Adam-Optimization.md]: included pytorch example and codebyte in the document.

**Comment by TimLyons87** [comment]
@mamtawardhani I have reviewed and included the requested changes to the Adam Optimization document. It is now ready for your review. Thank you.

**Review by mamtawardhani — APPROVED** [review]
Thank you for making the required changes @TimLyons87! 😄 

It looks good for a second review!

**Comment by ishg-153** [comment]
Hey @TimLyons87! You need to sign the CLA. Please do so from the @CLAassistant comment.

**Comment by TimLyons87** [comment]
@ishg-153 @mamtawardhani I have signed the CLA assistant agreement. Thank you both for your patience. Please let me know if you find any other errors or have other edits. Thanks again, Tim.

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@TimLyons87 Sorry for the delayed review. This entry is too long to be a docs entry. You will have to reduce the length and details of the entry. Docs entries are expected to be bite-sized well-rounded explanations. Use [this similar entry](https://github.com/Codecademy/docs/blob/main/content/ai/concepts/neural-networks/terms/linear-activation-function/linear-activation-function.md) as a base for structure.

Also make the following change.

**Comment by ishg-153** [comment]
@TimLyons87 Will you be working on this entry?

**Comment by mamtawardhani** [comment]
Closing this PR due to inactivity of the author.

**Pull request closed without merging**